tensorflow/tools/ci_build/Dockerfile.rocm

```diff
@@ -71,7 +71,20 @@ ENV PATH="$ROCM_PATH/bin:${PATH}"
 ENV PATH="$OPENCL_ROOT/bin:${PATH}"
 
 # Add target file to help determine which device(s) to build for
-RUN bash -c 'echo -e "gfx803\ngfx900\ngfx906" >> /opt/rocm/bin/target.lst'
+RUN bash -c 'echo -e "gfx803\ngfx900\ngfx906" >> ${ROCM_PATH}/bin/target.lst'
+
+# Need to explicitly create the $ROCM_PATH/.info/version file to workaround what seems to be a bazel bug
+# The env vars being set via --action_env in .bazelrc and .tf_configure.bazelrc files are sometimes
+# not getting set in the build command being spawned by bazel (in theory this should not happen)
+# As a consequence ROCM_PATH is sometimes not set for the hipcc commands.
+# When hipcc incokes hcc, it specifies $ROCM_PATH/.../include dirs via the `-isystem` options
+# If ROCM_PATH is not set, it defaults to /opt/rocm, and as a consequence a dependency is generated on the
+# header files included within `/opt/rocm`, which then leads to bazel dependency errors
+# Explicitly creating the $ROCM_PATH/.info/version allows ROCM path to be set correrctly, even when ROCM_PATH
+# is not explicitly set, and thus avoids the eventual bazel dependency error.
+# The bazel bug needs to be root-caused and addressed, but that is out of our control and may take a long time
+# to come to fruition, so implementing the workaround to make do till then
```

Comment:
Could you please reference the bazel issue that you filed here?

Reply:
@chsigg I have not filed a bazel issue for this, but I can. The only way I know to reproduce this issue, is to do a `--config=rocm` build with the `bazel -s` option, save the output to file, and then scan through the subcommand invocations and check whether all the `action_env`s specified on the bazel command were passes through correctly or not. Will that suffice?

Reply:
filed issue https://github.com/bazelbuild/bazel/issues/11163, and updated the comment to mention it

Reply:
@chsigg gentle ping

Reply:
@chsigg gentle ping